consensus/parlia/snapshot.go

```diff
@@ -262,16 +297,22 @@ func (s *Snapshot) apply(headers []*types.Header, chain consensus.ChainHeaderRea
 		if _, ok := snap.Validators[validator]; !ok {
 			return nil, errUnauthorizedValidator(validator.String())
 		}
-		for _, recent := range snap.Recents {
-			if recent == validator {
+		if chainConfig.IsBohr(header.Number, header.Time) {
+			if snap.SignRecently(validator) {
 				return nil, errRecentlySigned
 			}
+		} else {
```

Comment:
maybe we don't need this if/else, as `snap.SignRecently(validator)` also works for turnLength == 1?

Reply:
if snap.Recents has the right length, then the `if/else` is not need.
but snap.Recents may contain more elements than need, as happened [here](https://github.com/bnb-chain/bsc/pull/1344/files)